gameswap-service/src/main/java/org/gameswap/web/authentication/JwtAuthFilter.java

```diff
@@ -0,0 +1,100 @@
+package org.gameswap.web.authentication;
+
+import com.google.common.base.Optional;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jwt.JWTClaimsSet;
+
+import org.apache.commons.lang3.StringUtils;
+import org.gameswap.model.User;
+import org.gameswap.model.UserPrincipal;
+import org.gameswap.persistance.UserDAO;
+import org.gameswap.web.authentication.AuthUtils;
+import org.hibernate.CacheMode;
+import org.hibernate.FlushMode;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.context.internal.ManagedSessionContext;
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.text.ParseException;
+
+import javax.annotation.Priority;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.SecurityContext;
+
+@Priority(1000)
+public class JwtAuthFilter implements ContainerRequestFilter {
+    private UserDAO dao;
+    private SessionFactory sessionFactory;
+
+    public JwtAuthFilter(UserDAO dao, SessionFactory sessionFactory) {
+        this.dao = dao;
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        String authHeader = requestContext.getHeaderString(AuthUtils.AUTH_HEADER_KEY);
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            session.setDefaultReadOnly(true);
+            session.setCacheMode(CacheMode.NORMAL);
+            session.setFlushMode(FlushMode.MANUAL);
+            ManagedSessionContext.bind(session);
+            doWork(requestContext, authHeader);
+        }finally{
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    private void doWork(final ContainerRequestContext requestContext, String authHeader) {
+        if (!StringUtils.isBlank(authHeader) && authHeader.split(" ").length == 2) {
+            JWTClaimsSet claimSet;
+            try {
+                claimSet = AuthUtils.decodeToken(authHeader);
+            } catch (ParseException | JOSEException e) {
+                throw new NotAuthorizedException(e);
+            }
+
+            // ensure that the token is not expired
+            if (!new DateTime(claimSet.getExpirationTime()).isBefore(DateTime.now())) {
+                Optional<User> userOptional = dao.find(Long.parseLong(claimSet.getSubject()));
+                Optional<UserPrincipal> transform = userOptional.transform(user -> new UserPrincipal(user.getUsername()));
```

Comment:
I just used the existing code, I think our principal should be based on user id because usernames are no longer mandatory with the google signin.